src/ReactiveUI.Blazor/ReactiveComponentBase.cs

```diff
@@ -37,8 +35,7 @@ public class ReactiveComponentBase<T> : ComponentBase, IViewFor<T>, INotifyPrope
         /// </summary>
         public ReactiveComponentBase()
         {
-            this.WhenAnyValue(x => x.ViewModel).Subscribe(_ => StateHasChanged());
-            var viewModelsPropertyChanged = this.WhenAnyValue(x => x.ViewModel)
+            var viewModelsChangedObservable = this.WhenAnyValue(x => x.ViewModel)
```

Comment:
To keep the pr precise let's go back to the old ordering and variable names. Eg the view model change first and viewModelsPropertyChanged